feat: add implementation of `math/base/special/minmaxabsf`

sahil20021008 opened the pull request:
Resolves #6163 .

## Description

> What is the purpose of this pull request?

This pull request proposes adding minmaxabsf implementation

-   This pull request adds minmaxabsf implementation to stdlib. This includes benchmarks, lib files, tests, examples, src files, README, etc

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   Progresses #649 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

sahil20021008 committed:
feat!: add implementation of stdlib/math/base/special/minmaxabsf

This commit adds the implementation of stdlib/math/base/special/minmaxabsf function.
It includes a README.md file, benchmarks, tests, examples.
The minmaxabsf function returns the minimum and maximum single-precision floating-point values for any two numbers.

Closes: #6163

sahil20021008 committed:
feat!: add implementation of stdlib/math/base/special/minmaxabsf

This commit adds the implementation of stdlib/math/base/special/minmaxabsf function.
It includes a README.md file, benchmarks, tests, examples.
The minmaxabsf function returns the minimum and maximum single-precision floating-point values for any two numbers.

Closes: #6163

Neerajpathak07 commented on etc/npm/deps.txt:
```diff
@@ -1 +1 @@
-acorn@"^8.1.0" acorn-loose@"^8.0.2" acorn-walk@"^8.0.2" d3-format@"^1.0.0" d3-scale@"^1.0.0" d3-shape@"^1.0.0" d3-time-format@"^2.0.0" debug@"^2.6.9" glob@"^7.0.5" minimist@"^1.2.0" readable-stream@"^2.1.4" resolve@"^1.1.7" vdom-to-html@"^2.3.0" virtual-dom@"^2.1.1" 0x@"^4.10.2" @cspell/eslint-plugin@"^8.8.0" @commitlint/cli@"^17.4.4" @commitlint/cz-commitlint@"^17.4.4" @conventional-commits/pars ...
```
I believe this change is not desired for this PR.
Did you add this change after running `npm install` cause it surely looks like it.

Neerajpathak07 commented on package.json:
```diff
@@ -1,296 +1 @@
-{
```
same for this file as well.

sahil20021008 committed:
feat!: add implementation of stdlib/math/base/special/minmaxabsf

This commit adds the implementation of stdlib/math/base/special/minmaxabsf function.
It includes a README.md file, benchmarks, tests, examples.
The minmaxabsf function returns the minimum and maximum single-precision floating-point values for any two numbers.

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: passed
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: missing_dependencies
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

sahil20021008 committed:
feat!: add implementation of stdlib/math/base/special/minmaxabsf

This commit adds the implementation of stdlib/math/base/special/minmaxabsf function.
It includes a README.md file, benchmarks, tests, examples.
The minmaxabsf function returns the minimum and maximum single-precision floating-point values for any two numbers.

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

sahil20021008 committed:
feat!: add implementation of stdlib/math/base/special/minmaxabsf

This commit adds the implementation of stdlib/math/base/special/minmaxabsf function.
It includes a README.md file, benchmarks, tests, examples.
The minmaxabsf function returns the minimum and maximum single-precision floating-point values for any two numbers.

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: passed
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

sahil20021008 committed:
feat!: add implementation of stdlib/math/base/special/minmaxabsf

This commit adds the implementation of stdlib/math/base/special/minmaxabsf function.
It includes a README.md file, benchmarks, tests, examples.
The minmaxabsf function returns the minimum and maximum single-precision floating-point values for any two numbers.

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

kgryte commented on lib/node_modules/@stdlib/math/base/special/minmaxabsf/manifest.json:
```diff
@@ -0,0 +1,98 @@
+      "libraries": [],
+      "libpath": [],
+      "dependencies": [
+        "@stdlib/napi/argv",
+        "@stdlib/napi/export",
+        "@stdlib/napi/argv-float",
+        "@stdlib/napi/argv-float32array",
```
These deps should not be here.

kgryte commented on lib/node_modules/@stdlib/math/base/special/minmaxabsf/manifest.json:
```diff
@@ -0,0 +1,98 @@
+        "@stdlib/math/base/napi/binary",
+        "@stdlib/math/base/special/absf",
+        "@stdlib/math/base/assert/is-nanf",
+        "@stdlib/constants/float32/nan.h"
```
```suggestion
        "@stdlib/constants/float32/nan"
```

kgryte commented on lib/node_modules/@stdlib/math/base/special/minmaxabsf/manifest.json:
```diff
@@ -0,0 +1,98 @@
+      "libraries": [],
+      "libpath": [],
+      "dependencies": [
+        "@stdlib/napi/argv",
+        "@stdlib/napi/export",
+        "@stdlib/napi/argv-float",
+        "@stdlib/napi/argv-float32array",
```
These should be removed.

kgryte commented on lib/node_modules/@stdlib/math/base/special/minmaxabsf/manifest.json:
```diff
@@ -0,0 +1,98 @@
+        "@stdlib/math/base/napi/binary",
+        "@stdlib/math/base/special/absf",
+        "@stdlib/math/base/assert/is-nanf",
+        "@stdlib/constants/float32/nan.h"
```
```suggestion
        "@stdlib/constants/float32/nan"
```

kgryte commented on lib/node_modules/@stdlib/math/base/special/minmaxabsf/manifest.json:
```diff
@@ -0,0 +1,98 @@
+        "@stdlib/math/base/napi/binary",
+        "@stdlib/math/base/special/absf",
+        "@stdlib/math/base/assert/is-nanf",
+        "@stdlib/constants/float32/nan.h"
```
```suggestion
        "@stdlib/constants/float32/nan"
```

sahil20021008 committed:
feat!: Committing changes for draft review

anandkaranubc commented:
/stdlib merge

anandkaranubc commented:
@sahil20021008 I left some comments on Gitter. Just referencing them here.

"It's because you updated your local branch with the latest changes from the remote `develop` branch, and now your PR includes all those new changes as well.

You need to rebase/merge your branch onto the latest `develop` to clean up the PR. That'll make sure only your actual changes show up in the diff.

[Git guide](https://github.com/stdlib-js/stdlib/blob/c22efc37e82438a65bad53c133bcd78b504f585f/docs/contributing/git_cheatsheet.md)

You seem to have also modified some files like `math/base/special/cbrtf` which should be reverted." 

And that's the reason why the above bot command didn't work.

sahil20021008 committed:
Resolve merge conflicts

sahil20021008 committed:
feat: fix manifest.json errors

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

sahil20021008 commented:
/stdlib help

sahil20021008 commented:
/stdlib lint-autofix

stdlib-bot committed:
fix: resolve lint errors

sahil20021008 committed:
feat: fixed copyright dates

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: passed
  - task: lint_c_benchmarks
    status: passed
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: passed
  - task: lint_license_headers
    status: passed
---

sahil20021008 committed:
feat: fix julia benchmark

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

sahil20021008 committed:
feat: fix julia benchmark

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

sahil20021008 committed:
feat: fix julia benchmark

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

kgryte commented on lib/node_modules/@stdlib/math/base/special/minmaxabsf/README.md:
```diff
@@ -0,0 +1,258 @@
+
+-   **x**: `[in] float` input value.
+-   **y**: `[in] float` input value.
+-   **min**: `[in] *float` output min.
```
```suggestion
-   **min**: `[out] *float` output min.
```

sahil20021008 replied:
Done.

kgryte commented on lib/node_modules/@stdlib/math/base/special/minmaxabsf/README.md:
```diff
@@ -0,0 +1,258 @@
+-   **x**: `[in] float` input value.
+-   **y**: `[in] float` input value.
+-   **min**: `[in] *float` output min.
+-   **max**: `[in] *float` output max.
```
```suggestion
-   **max**: `[out] *float` output max.
```

sahil20021008 replied:
Done

kgryte commented on lib/node_modules/@stdlib/math/base/special/minmaxabsf/README.md:
```diff
@@ -0,0 +1,258 @@
+float min;
+float max;
+stdlib_base_minmaxabsf( -4.2f, 3.14f, &min, &max );
+// [ 3.14, 4.2 ]
```
These return annotations need to be fixed.

sahil20021008 replied:
Removed

kgryte commented on lib/node_modules/@stdlib/math/base/special/minmaxabsf/README.md:
```diff
@@ -0,0 +1,258 @@
+// [ 3.14, 4.2 ]
+
+stdlib_base_minmaxabsf( 0.0f, -0.0f, &min, &max );
+// [ 0.0, 0.0 ]
```
This needs to be fixed.

sahil20021008 replied:
Removed

kgryte commented on lib/node_modules/@stdlib/math/base/special/minmaxabsf/README.md:
````diff
@@ -0,0 +1,258 @@
+<!-- eslint no-undef: "error" -->
+
+```javascript
+var randu = require( '@stdlib/random/base/randu' );
````
We can refactor this example to use `random/array/uniform` and `console/log-each-map`.

sahil20021008 replied:
Done

kgryte commented on lib/node_modules/@stdlib/math/base/special/minmaxabsf/examples/c/example.c:
```diff
@@ -0,0 +1,36 @@
+#include <stdio.h>
+
+int main( void ) {
+	float x;
```
Order according to declaration length.

sahil20021008 replied:
Done

kgryte commented on lib/node_modules/@stdlib/math/base/special/minmaxabsf/examples/index.js:
```diff
@@ -0,0 +1,34 @@
+
+'use strict';
+
+var randu = require( '@stdlib/random/base/randu' );
```
Same comment as README

sahil20021008 replied:
Done

kgryte commented on lib/node_modules/@stdlib/math/base/special/minmaxabsf/manifest.json:
```diff
@@ -0,0 +1,79 @@
+        "@stdlib/napi/export",
+        "@stdlib/napi/argv-float",
+        "@stdlib/napi/argv-float32array",
+        "@stdlib/math/base/napi/quaternary",
```
Why do you have this dep?

sahil20021008 replied:
This dependencies are being used in src/addon.c file

kgryte reviewed (changes requested):
Left an initial round of comments.

sahil20021008 committed:
feat: addressed review comments

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: passed
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

Planeshifter commented:
<!-- stdlib-bot-tracking-issue-closure-check -->
:warning: **Tracking Issue Closure Warning** :warning:

I noticed your PR description contains closing keywords ("Resolves", "Closes", or "Fixes") referencing a "Tracking Issue".

**Why this matters:**
Tracking issues should typically remain open until all related sub-issues are completed. GitHub automatically closes issues with such closing keywords when the PR is merged. For more information, see [GitHub's documentation on using keywords in issues and pull requests][github-keywords].

**Required action:**
Use "Progresses" instead to reference the tracking issue without automatically closing it.

Thank you for your contribution to the project!

[github-keywords]: https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests

sahil20021008 commented:
I have changed the PR description as requested.
I have changed "resolves" keywords to "progresses"

sahil20021008 committed:
feat: add implementation of stdlib/math/base/special/minmaxabsf

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc commented:
Completed here: https://github.com/stdlib-js/stdlib/pull/7043
